File: crowdin/src/main/java/com/crowdin/platform/transformers/NavigationViewTransformer.java
package com.crowdin.platform.transformers;

import android.support.design.widget.NavigationView;
import android.view.Menu;
import android.view.View;

/**
 * A transformer which transforms NavigationView: it transforms the texts coming from the menu.
 */
public class NavigationViewTransformer extends BaseNavigationViewTransformer {

    @Override
    public Class<? extends View> getViewType() {
        return NavigationView.class;
    }

    @Override
    protected Menu getMenu(View view) {
        return ((NavigationView) view).getMenu();
    }
}

File: crowdin/src/main/java/com/crowdin/platform/transformers/NavigationViewTransformer.kt
package com.crowdin.platform.transformers

import android.support.design.widget.NavigationView
import android.view.Menu
import android.view.View

/**
 * A transformer which transforms NavigationView: it transforms the texts coming from the menu.
 */
internal class NavigationViewTransformer : BaseNavigationViewTransformer() {

    override val viewType: Class<out View>
        get() = NavigationView::class.java

    override fun getMenu(view: View): Menu {
        return (view as NavigationView).menu
    }
}
